src/main/java/com/impactupgrade/nucleus/client/PaypalClient.java

```diff
@@ -85,10 +84,18 @@ public static final class SellerReceivableBreakdown {
     public Amount grossAmount;
     @JsonProperty("paypal_fee")
     public Amount paypalFee;
+    @JsonProperty("platform_fees")
+    public List<PlatformFee> platformFees;
```

Comment:
Was this (and the removal of `payee`) the only reason the Capture event wasn't validating against the signature? Any other concerns with using the old/deprecated library? Although, we're minimizing our usage of it...

Reply:
Event validation is using request body/headers and webhook id to validate the request. (In my case was working ok with both versions of Capture model).
Event deserialization happens after validation. (Was failing bcz payee was not deserialized correctly which caused NPE later on).
Payee removal (moving it to platform fees list) fixed a problem with the deserialization (but has not affected validation in any way).

I agree with the point that we should avoid using deprecated lib. We can replace lib's models with our own ones.
One big piece that lib has useful for us is managing auth (in case of removing the lib we will have to manually update oauth tokens). But mb we can also use our own classes (OAuthContext?) to manage this part.

Reply:
Event validation is using request body/headers and webhook id to validate the request. (In my case was working ok with both versions of Capture model).
Event deserialization happens after validation. (Was failing bcz payee was not deserialized correctly which caused NPE later on).
Payee removal (moving it to platform fees list) fixed a problem with the deserialization (but has not affected validation in any way).

I agree with the point that we should avoid using deprecated lib. We can replace lib's models with our own ones.
One big piece that lib has useful for us is managing auth (in case of removing the lib we will have to manually update oauth tokens). But mb we can also use our own classes (OAuthContext?) to manage this part.